authz/sdk_end2end_test.go

```diff
@@ -518,29 +533,45 @@ func (s) TestSDKFileWatcher_RecoversFromReloadFailure(t *testing.T) {
 	}
 
 	// Skips the invalid policy update, and continues to use the valid policy.
-	if err := ioutil.WriteFile(file, []byte("{}"), os.ModePerm); err != nil {
-		t.Fatalf("ioutil.WriteFile(%q) failed: %v", file, err)
+	if err := os.WriteFile(file, []byte("{}"), os.ModePerm); err != nil {
+		t.Fatalf("os.WriteFile(%q) failed: %v", file, err)
 	}
-	// Wait 2 seconds for background go routine to read the updated files.
-	time.Sleep(2 * time.Second)
+	// Wait 30 ms for background go routine to read the updated files.
+	time.Sleep(30 * time.Millisecond)
 
 	// Verifying authorization decision.
 	_, err = client.UnaryCall(ctx, &pb.SimpleRequest{})
-	if got := status.Convert(err); got.Code() != valid1.wantStatusCode || got.Message() != valid1.wantErr {
-		t.Fatalf("[UnaryCall] error want:{%v %v} got:{%v %v}", valid1.wantStatusCode, valid1.wantErr, got.Code(), got.Message())
+	numRetries := 0
```

Comment:
This pattern repeats several times.  Factor it into a function?

Reply:
So the pattern for when we skip an invalid update, and when we use a valid update is slightly different. This particular snippet appears twice in the code. I feel using a different function here may reduce the readability of code. The function would need to accept several parameters TestServiceClient, context, status to compare against. Also if we don't want to pass testing.T to this function, that would require us to check for return value, and then fail the test accordingly. So I don't see much benefit in the refactoring.

Reply:
I think this would be great:

```go
func retryUntil(ctx context.Context, tsc pb.TestServiceClient, want *status.Status) (lastErr error) {
	for ctx.Err() == nil {
		lastErr = tsc.UnaryCall(ctx, &pb.SimpleRequest{})
		if s := status.Convert(lastErr); s.Code() == want.Code() && s.Message() == want.Message() {
			return nil
		}
		time.Sleep(whatever*time.Millisecond)
	}
	return lastErr
}
```

This is short, clear, and reusable.  Usage:

```go
if got := retryUntil(ctx, client, valid1.wantStatus); got != nil {
	t.Fatalf("want, got, etc...", valid1.wantStatus.Err(), got)
}
```

Reply:
Sure. Updated code. Please note that there are two separate functions here, for when we skip a reload and when we use a valid reload.